android/pytorch_android/build.gradle

```diff
@@ -85,6 +85,12 @@ task sourcesJar(type: Jar) {
 }
 
 def getLibtorchHeadersDir() {
+  println("hasProperty(LIBTORCH_HEADERS):" + project.hasProperty('LIBTORCH_HEADERS'))
+  if (project.hasProperty('LIBTORCH_HEADERS')) {
+    println("LIBTORCH_HEADERS:" + property('LIBTORCH_HEADERS'))
+    return property('LIBTORCH_HEADERS')
+  }
+
   def abi = ABI_FILTERS.split(",")[0]
   return "$rootDir/../build_android_$abi/install/include";
```

Comment:
When do we hit this default value? Is it for `scripts/build_pytorch_android.sh`?

Can we always pass in this directory explicitly? When I refactored the script recently, I added the ability to override the `BUILD_ROOT` for build_pytorch_android.sh, i.e. `build_android_$abi` can now be in places like pytorch/build_pytorch_android_custom/build_android_x86/... - see my PR #40199 

In general I feel it would be more robust to reduce the hardcoded path constants in codebase :)

Reply:
Yes, default value is for local build with `scripts/build_pytorch_android.sh`, that will also needed for `bintrayUpload` task for release publishing.

Yes, I agree, with hardcoding of paths we have to integrate gradle changes twice:

As we have different paths locally and on CI, on CI we have variations like `~/workspace/build_android_install_$abi/install/include` (_install_ in the middle) which is copied from abi docker image.

For selective build we do not have exported functions in binarys and we do not need headers in the package, am I right that custom build is the same as selective build that we also do not need to add headers inside?

Reply:
I will try to remove hardcoded path for local build using the same -PLIBTORCH_HOME